VSCode/armsnippets.json

```diff
@@ -506,6 +506,145 @@
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"networkRuleCollections\": [",
+        "           {",
+        "              \"name\": \"${22:networkCollectionName}\",",
+        "               \"properties\": {",
+        "                   \"priority\": \"${23|100,100-65000|}\",",
+        "                   \"action\": {",
+        "                      \"type\": \"${24|Deny,Allow|}\"",
+        "                   },",
+        "                   \"rules\": [",
+        "                      {",
+        "                          \"name\": \"${25:networkRuleName}\",",
+        "                          \"description\": \"${26:networkRuleDescription}\",",
+        "                          \"sourceAddresses\": [",
+        "                             \"${27:10.0.0.4}\"",
+        "                          ],",
+        "                          \"destinationAddresses\": [",
+        "                              \"${28:10.0.1.4}\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${29|1,1-64000,*|}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${30|TCP,UDP,Any,ICMP|}\"",
+        "                          ]",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"ipConfigurations\": [",
+        "           {",
+        "               \"name\": \"${31:ipConfigName}\",",
+        "               \"properties\": {",
+        "                   \"subnet\": {",
+        "                       \"id\": \"${32:[resourceId('virtualNetworkResourceGroup', 'Microsoft.Network/virtualNetworks/subnets', 'virtualNetworkName', 'AzureFirewallSubnet')]}\"",
+        "                   },",
+        "                   \"publicIPAddress\": {",
+        "                       \"id\": \"${33:[resourceId('Microsoft.Network/publicIPAddresses', '${34:publicIPAddress1}')]}\"",
+        "                   }",
+        "               }",
+        "           }",
+        "       ]",
+        "     },",
+        "    \"dependsOn\": [",
+        "        \"[resourceId('Microsoft.Network/publicIPAddresses', '${34:publicIPAddress1}')]\"",
+        "    ]",
+        "},",
+        "{",
+        "    \"name\": \"${34:publicIPAddress1}\",",
+        "    \"type\": \"Microsoft.Network/publicIPAddresses\",",
+        "    \"apiVersion\": \"2019-04-01\",",
+        "    \"location\": \"[resourceGroup().location]\",",
+        "    \"sku\": {",
+        "       \"name\": \"standard\"",
+        "       },",
+        "    \"tags\": {",
+        "        \"displayName\": \"${1:firewall1}\"",
```

Comment:
\"displayName\": \"${34:publicIPAddress1}\"",

Reply:
I'd prefer to tag the public IP in this case with the device it is associated with, in this case the firewall.